core/jobs/src/main/java/com/cognifide/aet/job/common/modifiers/replacetext/ReplaceTextModifier.java

```diff
@@ -0,0 +1,99 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.cognifide.aet.job.common.modifiers.replacetext;
+
+import com.cognifide.aet.communication.api.metadata.CollectorStepResult;
+import com.cognifide.aet.job.api.collector.CollectorJob;
+import com.cognifide.aet.job.api.collector.CollectorProperties;
+import com.cognifide.aet.job.api.exceptions.ParametersException;
+import com.cognifide.aet.job.api.exceptions.ProcessingException;
+import com.cognifide.aet.job.common.modifiers.WebElementsLocatorParams;
+import org.apache.commons.lang3.StringUtils;
+import org.openqa.selenium.JavascriptExecutor;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebDriverException;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.By;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.Map;
+
+public class ReplaceTextModifier extends WebElementsLocatorParams implements CollectorJob {
+
+    public static final String NAME = "replaceText";
+
+    private static final Logger LOG = LoggerFactory.getLogger(ReplaceTextModifier.class);
+
+    private static final String ATTRIBUTE_PARAM = "attributeName";
+    private static final String VALUE_PARAM = "value";
+
+    private final WebDriver webDriver;
+
+    private final CollectorProperties properties;
+
+    private String attributeName;
+    private String value;
+
+    public ReplaceTextModifier(WebDriver webDriver, CollectorProperties properties) {
+        this.webDriver = webDriver;
+        this.properties = properties;
+    }
+
+    @Override
+    public CollectorStepResult collect() throws ProcessingException {
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Replacing element for {}: {} on page: {} with attributeName : {} and value: {}. url: {}",
+                    getSelectorType(), getSelectorValue(), webDriver.getCurrentUrl(), attributeName, value, properties.getUrl(),
+                    properties.getUrl());
+        }
+        return replaceText();
+    }
+
+    @Override
+    public void setParameters(Map<String, String> params) throws ParametersException {
+        setElementParams(params);
+        attributeName = StringUtils.defaultIfBlank(params.get(ATTRIBUTE_PARAM), "innerHTML");
+        value = StringUtils.defaultIfBlank(params.get(VALUE_PARAM), "");
+    }
+
+    public CollectorStepResult replaceText() throws ProcessingException {
+        CollectorStepResult result;
+        try {
+            String script = "arguments[0]." + attributeName + "=arguments[1];";
+
+            By elementLocator = getLocator();
+            waitForElementToBePresent(webDriver, elementLocator);
+
+            List<WebElement> webElements = webDriver.findElements(elementLocator);
```

Comment:
Maybe we could separate finding an element and updating text into two separate methods with separate try-catch blocks? (nice to have)